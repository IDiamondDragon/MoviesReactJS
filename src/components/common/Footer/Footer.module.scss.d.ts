declare namespace FooterModuleScssNamespace {
  export interface IFooterModuleScss {
    footer: string;
    footer__baner: string;
  }
}

declare const FooterModuleScssModule: FooterModuleScssNamespace.IFooterModuleScss & {
  /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
  locals: FooterModuleScssNamespace.IFooterModuleScss;
};

export = FooterModuleScssModule;
