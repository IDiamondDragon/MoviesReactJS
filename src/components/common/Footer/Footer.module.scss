@import "../../../styles/supporting/colors";

.footer {
    position: relative;
    padding: 16px;
    background-color: $tundora; 

    &__baner {
        display: block;
        margin: 0 auto;
    }
}

