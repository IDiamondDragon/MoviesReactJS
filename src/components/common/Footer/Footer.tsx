import React from "react";
import Baner from '../../common/Banner/Baner';

import styles from './Footer.module.scss';



export function Footer(): JSX.Element {
  return (
    <div className={styles.footer}>
      <Baner className={styles.footer__baner} />
    </div>
  );
}

export default React.memo(Footer);