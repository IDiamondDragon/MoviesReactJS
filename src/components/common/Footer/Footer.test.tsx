import React from 'react';
import { render } from '@testing-library/react';
import Footer from './Footer';


jest.mock('../../../services/hooks/useRedirectToSearchPage', () => {
  return {
      useRedirectToSearchPage: jest.fn().mockImplementation(() => {
      return () => ({});
    }),
  };
});


describe('<Footer />', () => {

  test("renders correctly", () => {
      const { asFragment } = render(<Footer />);

      expect(asFragment()).toMatchSnapshot();
  })

});
