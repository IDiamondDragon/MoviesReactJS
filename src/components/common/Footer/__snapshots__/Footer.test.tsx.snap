// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="footer"
  >
    <img
      class="baner footer__baner"
      src="test-file-stub"
    />
  </div>
</DocumentFragment>
`;
